config/smarty.config.inc.php

```diff
@@ -177,9 +177,7 @@ function smartyCleanHtml($data)
 function smartyClassname($classname)
 {
     $classname = Tools::replaceAccentedChars(strtolower($classname));
-    $classname = preg_replace('/[^A-Za-z0-9]/', '-', $classname);
-    $classname = preg_replace('/[-]+/', '-', $classname);
-
+    $classname = preg_replace(array("/[^A-Za-z0-9-_]/", "/[-]{3,}/", "/[-]+$/"), array("-", '-', '') , $classname);
```

Comment:
I don't understand why `"/[-]{3,}/"`? what do you mean? :thinking:

Reply:
To my knowledge, there is no CSS methodology that adds more than two dash. And I don't think it ever exists for web performance or reading issues.
I have already come across CSS methods that use the double dash at the beginning of the class name, but not the end.

The second ReGex makes it possible to remove the dashes when there are 3 or more consecutive dashes.
The third Regex allows you to remove the double end dashes that will remain in some cases.

If we take the example of the documentation:
`Hérè-Is_a-Clàssnåme--@#$$ˆ*(&-----`

- For the first regex, this gives: `H-r--Is_a-Cl-ssn-me---------------`
- For the second ReGex: `H-r--Is_a-Cl-ssn-me`
- The third ReGex in this case will do nothing.

But if we take an example with special characters distributed differently, for example a deliberately false character string: `<input type="text" name="firstname">`

- For the first regex, this gives: `-input-type--text--name--firstname--`
- For the second ReGex (No change): `-input-type--text--name--firstname--`
- For the third ReGex (It cleans the end): `-input-type--text--name--firstname`

In theory, the developer of the module or theme should see that there is a problem with their class name.